x/launch/keeper/msg_server_settle_request_test.go

```diff
@@ -0,0 +1,268 @@
+package keeper
+
+import (
+	"fmt"
+	"testing"
+
+	codec "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/stretchr/testify/require"
+	spnerrors "github.com/tendermint/spn/pkg/errors"
+	"github.com/tendermint/spn/testutil/sample"
+	"github.com/tendermint/spn/x/launch/types"
+)
+
+func TestMsgSettleRequest(t *testing.T) {
+	var (
+		invalidChain, _          = sample.ChainID(0)
+		addr1                    = sample.AccAddress()
+		k, pk, srv, _, sdkCtx, _ = setupMsgServer(t)
+		ctx                      = sdk.WrapSDKContext(sdkCtx)
+		chains                   = createNChain(k, sdkCtx, 2)
+		requests                 = createNRequest(k, sdkCtx, 6)
+	)
+	chains[0].LaunchTriggered = true
+	k.SetChain(sdkCtx, chains[0])
+	chains[1].ChainID = "foo"
+	k.SetChain(sdkCtx, chains[1])
+	tests := []struct {
+		name string
+		msg  types.MsgSettleRequest
+		want uint64
+		err  error
+	}{
+		{
+			name: "invalid chain",
+			msg: types.MsgSettleRequest{
+				ChainID:     invalidChain,
+				Coordinator: addr1,
```

Comment:
Can we use `pk.GetCoordinatorAddressFromID(sdkCtx, chains[1].CoordinatorID)` as well if the test is not about invalid coordinator

Reply:
We don't have the chain list object for this method. We don't need them. The chain validations regarding coordinator address are in the parent method. This one is only for apply the request. I created an object to represent the coordinator account in the tests to be easier